lnwallet/channel.go

```diff
@@ -6916,16 +6913,52 @@ func (lc *LightningChannel) MaxFeeRate(maxAllocation float64,
 	// Ensure the fee rate doesn't dip below the fee floor.
 	_, weight := lc.availableBalance()
 	maxFeeRate := maxFee / (float64(weight) / 1000)
-	feeRate := chainfee.SatPerKWeight(
+	return chainfee.SatPerKWeight(
 		math.Max(maxFeeRate, float64(chainfee.FeePerKwFloor)),
 	)
+}
+
+// IdealCommitFee returns uses the current network fee, the minimum relay fee,
+// maximum fee allocation and anchor channel commitment fee rate to determine
+// the ideal fee to be used for the commitments of the channel.
+func (lc *LightningChannel) IdealCommitFee(netFee, minRelayFee,
+	maxAnchorCommitFeeRate chainfee.SatPerKWeight,
+	maxFeeAlloc float64) chainfee.SatPerKWeight {
+
+	// We'll cap the fee rate to our max fee allocation.
+	maxFee := lc.maxFeeRate(maxFeeAlloc)
+
+	commitFee := chainfee.SatPerKWeight(
+		math.Min(float64(netFee), float64(maxFee)),
+	)
+
+	// If the channel has anchor outputs then we cap the fee at the max
+	// anchor commitment fee rate.
+	if lc.channelState.ChanType.HasAnchors() &&
+		maxFee > maxAnchorCommitFeeRate {
```

Comment:
On another pass the shift makes more sense, as since we may shift our fee rate up (above the max anchor fee), we need to be able to respect the current configured max fee allocation. However this begs the question of what we should do if the max fee allocation is actually _below_ the min relay fee rate. I think this would be pretty rare in practice, unless someone "misconfigured" their channel fee related settings.

Reply:
I think `maxFee` here should be `commitFee` as well. Generally I think this function can be simplified somewhat.

Reply:
I don't think this would be rare, maybe unexpected but it could happen depending on the local balance, dust limit satoshis, and the min relay fee.